src/render/OpenGL.cpp

```diff
@@ -1468,7 +1471,19 @@ void CHyprOpenGLImpl::passCMUniforms(const CShader& shader, const NColorManageme
     glUniform1f(shader.sdrBrightness,
                 modifySDR && m_RenderData.pMonitor->sdrBrightness > 0 && targetImageDescription.transferFunction == NColorManagement::CM_TRANSFER_FUNCTION_ST2084_PQ ?
                     m_RenderData.pMonitor->sdrBrightness :
+
                     1.0f);
+    const auto cacheKey = std::make_pair(imageDescription.getId(), targetImageDescription.getId());
+    if (!primariesConversionCache.contains(cacheKey)) {
+        const auto                   mat             = imageDescription.getPrimaries().convertMatrix(targetImageDescription.getPrimaries()).mat();
+        const std::array<GLfloat, 9> glConvertMatrix = {
+            mat[0][0], mat[1][0], mat[2][0], //
+            mat[0][1], mat[1][1], mat[2][1], //
+            mat[0][2], mat[1][2], mat[2][2], //
+        };
+        primariesConversionCache.insert(std::make_pair(cacheKey, glConvertMatrix));
```

Comment:
nitpick: is there a reason you're doing `insert` and `std::make_pair` instead of just `primariesConversionCache[cacheKey] = glConvertMatrix;`?